README.md

```diff
@@ -0,0 +1,197 @@
+        pathlib.Path("/path/to/sdccc/result/directory"),
+    )
+
+    try:
+        return_code, direct_result, invariant_result = runner.run(  # https://github.com/Draegerwerk/SDCcc/?tab=readme-ov-file#exit-codes
+            config=pathlib.Path("/path/to/configuration/file.toml"),
+            requirements=pathlib.Path("/path/to/requirements/file.toml"),
+        )
+    except subprocess.TimeoutExpired:
+        print("Timeout occurred")
+        return
+
+    if direct_result is None or invariant_result is None:
+        print("No result file available")
+        return
+
+    for test_case in direct_result + invariant_result:
+        print(f"{test_case.name}: {test_case.is_passed}")
+```
+If you look for an async version
+
+```python
+import pathlib
+
+import pysdccc
+
+
+async def main():
+    if not await pysdccc.ais_downloaded("my-specific-version"):
+        await pysdccc.adownload("https://url/to/sdccc.zip")
+
+    runner = pysdccc.SdcccRunnerAsync(
+        pathlib.Path("/path/to/sdccc/result/directory"),
+    )
+
+    try:
+        return_code, direct_result, invariant_result = await runner.run(
+            # https://github.com/Draegerwerk/SDCcc/?tab=readme-ov-file#exit-codes
+            config=pathlib.Path("/path/to/configuration/file.toml"),
+            requirements=pathlib.Path("/path/to/requirements/file.toml"),
+        )
+    except TimeoutError:
+        print("Timeout occurred")
+        return
+
+    # checkout example from above ...
+```
+
+### Download an sdccc executable
+
+Check out `pysdccc.download` or `pysdccc.download_async`. Otherwise, if `pysdccc[cli]` is installed, `$ pysdccc install https://url/to/sdccc.zip` can also be used.
+
+### Create configuration file
+
+Configure the test consumer. Check the [test consumer configuration](https://github.com/Draegerwerk/SDCcc/?tab=readme-ov-file#test-consumer-configuration) for more information.
+
+```python
+import pathlib
+import toml  # has to be installed by the user
+import pysdccc
+
+config = {
+    'SDCcc': {
+        ...  # add all relevant config parameter
+    }
+}
+config_path = pathlib.Path('/path/to/configuration/file.toml')
+config_path.write_text(toml.dumps(config))
+
+runner = pysdccc.SdcccRunner(
+    pathlib.Path('/path/to/sdccc/result/directory'),
+)
+
+runner.run(
+    config=config_path,
+    requirements=pathlib.Path('/path/to/requirements/file.toml'),
+)
+
+# or if you have already downloaded the version
+config = runner.get_config()  # load default configuration
+config['SDCcc']['Consumer']['DeviceEpr'] = "urn:uuid:12345678-1234-1234-1234-123456789012"  # e.g. change device epr
+# save and run as above
+```
+
+### Create requirements file
+
+Enable or disable specific requirements. Check the [test requirements](https://github.com/Draegerwerk/SDCcc/?tab=readme-ov-file#enabling-tests) for more information.
+
+```python
+import pathlib
+import toml  # has to be installed by the user
+import pysdccc
+
+requirements = {
```

Comment:
as a user I immediately ask myself whether the reqirements dictionary can be extended by adding other keys for other core standards. Maybe rxpand your example

Reply:
ofc it can be extended, if SDCcc supports it. There is also a link to the tests supported by sdccc. I added a comment to explain what the key and what the value is